docs/branches.md

```diff
@@ -138,6 +138,17 @@ Branches that already existed before the `jj git fetch` are not affected. This
 is similar to Mercurial, which fetches all its bookmarks (equivalent to Git
 branches) by default.
 
+## Branch movement
+
+Currently Jujutsu only moves branches by itself when these conditions are met:
+
+ * When a commit has been rewritten (so when you `jj split` a change) the 
+   working-copy will move along with it.
+ * When a commit has been abandoned, all associated branches will be moved 
+   with the working-copy to its parent(s).
```

Comment:
I don't think working copy (or checked out revision) is important in this context. Maybe it's mentioned here as a comparison to Git's HEAD branch, but I think it's rather confusing.

Reply:
I thought it was meant to say that the working-copy commit will be moved to the parent(s), but I agree that "with the working-copy" is confusing because it makes it sound like it only happens if the branches point to a working-copy commit. I would write it something like this:

> When a commit has been abandoned, all associated branches will be moved to its parent(s). If a working copy was pointing to the abandoned commit, then a new working-copy commit will be created on top of the parent(s).

It's currently not true that we create the new working-copy commit on top of multiple parents if the abandoned commit was a merge commit, but I'm not sure it's worth mentioning that detail.

Reply:
rewritten to Martins suggestion